### ISSUE TITLE
Replace environment variables in jvm.options file.

### ISSUE THREAD

**Issue by codebrain** [elastic/elasticsearch#39090]
The `elasticsearch.bat` file replaces the `${ES_TMPDIR}` token with the environment variable stored in `!ES_TMPDIR!`:

https://github.com/elastic/elasticsearch/blob/dfecb256cb566de2913cd00a040e289f80040428/distribution/src/bin/elasticsearch.bat#L47

The `elasticsearch.exe` makes a similar replacement, as demonstrated here:

https://github.com/elastic/windows-installers/blob/b63a3e290c4cc4d90e5a3f90c160c5a5cdabebd9/src/ProcessHosts/Elastic.ProcessHosts/Elasticsearch/Process/ElasticsearchProcess.cs#L134

Can I suggest that that the `JvmOptionsParser` uses the environment variables to perform replacements in the `jvm.options` file?

This would allow the user to arbitrarily use environment variables as replacement values in the `jvm.options` file without having to manually configure those hard-coded replacements them in the `bat` or `exe` files. This would give users a little more flexibility when configuring their instances.

**Comment by jasontedor** [comment]
Over the years it would be that we have tried to greatly simplify our packaging. When we had `N > 1` ways of doing things, we tried to make it one. Where we had great flexibility to serve use cases at the edge, we removed them. We have done this effort all in the name of removing the surface area of bugs, and focusing on the most common uses of our software. This had pervaded the evolution of our packaging for several years now. This philosophy is important because packaging is very hard, and even harder to test the myriad of configurations and systems out there that the packaging has to interact with. Our `jvm.options` file is starting to get complex. It started as a simple text file where we merely appended all the non-comments into the command line flags for the JVM. We had to add support for multiple versions, otherwise we would be left unable to support JDK 9+ out of the box, or we would have to ship more than one configuration file; neither of those are palatable options (especially with more JDKs coming faster than ever). We had to add support for `ES_TMPDIR` because Linux can not get `/tmp` right for its life (it's not private by default); we needed a way to ensure we always use a private temporary directory, but how to tell `jvm.options` this since it has to set `-Djava.io.tmpdir`? When we add complexity, we try to confine it, and it is to solve real needs. We have been down the path of infinite flexibility before and there be dragons here.

I am not sure what enabling `jvm.options` to handle environment variables in general provides for us? It just moves some of the configuration to another file (well, how else are the environment variables going to get set?). And now we have a layer of indirection which is hard on everyone, when for the most part, we don't even think users should be messing much with `jvm.options`. Other than heap size, we try to ship defaults that are good for production use cases, and it should otherwise mostly be hands off.

I would prefer that we not do this.
